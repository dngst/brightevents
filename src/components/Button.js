import styled from "styled-components";

const Button = styled.button`
  color: ${(props) => (props.primary ? "#ffffff" : "#f78f1e")};
  background-color: ${(props) => (props.primary ? "#f78f1e" : "#ffffff")};
  border: 1px solid #f78f1e;
  border-radius: 0;
  margin: auto;
  padding: 0.7em 2em;
  font-size: 0.9em;
  text-transform: uppercase;
  font-weight: bold;
  cursor: pointer;
  width: ${(props) => (props.max ? "max-content" : "100%")};
`;

export default Button;
