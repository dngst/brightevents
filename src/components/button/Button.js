import styled from 'styled-components'

const Button = styled.button`
  background-color: #ffffff;
  border-radius: 3px;
  border: 1px solid #f78f1e;
  color: #f78f1e;
  margin: 0 1em;
  padding: 0.7em 2em;
  font-size: 0.9em;
`

export default Button;
