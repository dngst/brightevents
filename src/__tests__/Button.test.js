import { render } from "@testing-library/react";
import Button from "components/Button";

it("works", () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toHaveStyleRule("text-transform", "uppercase");
  expect(container.firstChild).toHaveStyleRule("border-radius", "0");
  expect(container.firstChild).toHaveStyleRule("font-size", "0.9em");
  expect(container.firstChild).toHaveStyleRule("font-weight", "bold");
  expect(container.firstChild).toHaveStyleRule("width", "100%");
  expect(container.firstChild).toHaveStyleRule("border", "1px solid");
});

it("applies default styles according to passed props", () => {
  const { container } = render(<Button max="true" primary />);
  expect(container.firstChild).toHaveStyleRule("color", undefined);
  expect(container.firstChild).toHaveStyleRule("width", "max-content");
});
