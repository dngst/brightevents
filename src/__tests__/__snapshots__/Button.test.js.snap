// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  border: 1px solid;
  border-radius: 0;
  margin: auto;
  padding: 0.7em 2em;
  font-size: 0.9em;
  text-transform: uppercase;
  font-weight: bold;
  cursor: pointer;
  width: 100%;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
}

<button
  class="c0"
/>
`;
